batch_job/src/batch_job_amazon_batch.c

```diff
@@ -0,0 +1,465 @@
+		fatal("credentials file does not contain aws_key");
+	if(!aws_region)
+		fatal("credentials file does not contain aws_reg");
+	if(!bucket_name)
+		fatal("credentials file does not contain bucket");
+	if(!queue_name)
+		fatal("credentials file does not contain queue_name");
+	if(!compute_env_name)
+		fatal("credentials file does not contain env_name");
+	if(!vpc)
+		fatal("credentials file does not contain vpc");
+	if(!subnet)
+		fatal("credentials file does not contain subnet"); 
+	
+	char* env_var = string_format("AWS_ACCESS_KEY_ID=%s AWS_SECRET_ACCESS_KEY=%s AWS_DEFAULT_REGION=%s ",aws_access_key_id,aws_secret_access_key,aws_region);
+	
+
+	initialized_data.aws_access_key_id = aws_access_key_id;
+	initialized_data.aws_secret_access_key = aws_secret_access_key;
+	initialized_data.aws_region=aws_region;
+	initialized_data.master_env_prefix = env_var;
+	return initialized_data;
+}
+
+static batch_job_id_t get_id_from_name(char* name){//name has structure of "QUEUENAME_id"
+	int i = strrpos(name,'_');//get last instance
+	int k = atoi(&(name[i+1]));
+	return k;
+}
+
+static char* generate_s3_cp_cmds(char* files, char* src, char* dst){
+	char* env_var = initialized_data.master_env_prefix;
+	int i;
+	int files_split_num = 0;
+	char** files_split;
+	split_comma_list(files, &files_split_num, &files_split);
+	char* new_cmd=malloc(sizeof(char)*1);
+        new_cmd[0]='\0';
+	if(files_split_num > 0){
+		char* copy_cmd_prefix = string_format("%s aws s3 cp ", env_var);
+		for(i=0; i<files_split_num; i++){
+			char* tmp = string_format("%s %s/%s %s",copy_cmd_prefix, src, files_split[i], dst);
+			char* tmp2 = string_format("%s\n%s\n",new_cmd,tmp);
+			free(new_cmd);
+			free(tmp);
+			new_cmd = tmp2;
+		}
+	}
+	clean_str_array(files_split_num,&files_split);
+	return new_cmd;
+}
+
+static void upload_cmd_file(char* bucket_name, char* input_files, char* output_files, char* cmd, int jobid){
+	char* env_var = initialized_data.master_env_prefix;
+	//Create command to pull files from s3 and into local space to work on
+	char* bucket = string_format("s3://%s",bucket_name);
+	char* cpy_in = generate_s3_cp_cmds(input_files,bucket,"./");
+	//run the actual command
+	char* cmd_tmp = string_format("%s\n%s\n",cpy_in,cmd);
+	free(cpy_in);
+	//copy out any external files
+	char* cpy_out = generate_s3_cp_cmds(output_files,"./",bucket);
+	cmd_tmp = string_format("%s\n%s\n",cmd_tmp,cpy_out);
+
+	//add headder
+	char* final_cmd = string_format("#!/bin/sh\n%s",cmd_tmp);
+	free(cmd_tmp);
+
+	//write out to file	
+	FILE* tmpfile = fopen("TEMPFILE.sh","w+");
+	fwrite(final_cmd,sizeof(char),strlen(final_cmd),tmpfile);
+	fclose(tmpfile);
+	free(final_cmd);
+	
+	//make executable and put into s3
+	system("chmod +x TEMPFILE.sh");
+	cmd_tmp = string_format("%s aws s3 cp ./TEMPFILE.sh s3://%s/COMAND_FILE_%i.sh",env_var,bucket_name,jobid);
+	system(cmd_tmp);
+	free(cmd_tmp);
+	remove("TEMPFILE.sh");	
+
+
+}
+
+static char* aws_submit_job(char* job_name, char* properties_string){
```

Comment:
Split this into two functions: one to define the job and the other to submit it.